File: server/src/utils/gemini.ts
import { GoogleGenAI } from "@google/genai";
const GEMINI_API_KEY = process.env.GEMINI_API_KEY;

const ai = new GoogleGenAI({ apiKey: GEMINI_API_KEY });

export async function analyzeResume(resumeText: string): Promise<{
  score: number;
  strengths: string[];
  improvements: string[];
  suggestions: string[];
  error?: string;
}> {
  const prompt = `
Analyze this resume for ATS (Applicant Tracking System) optimization. Provide:
1. A score out of 100
2. Key strengths
3. Areas for improvement
4. Suggested improvements.

All of these should be an array.

Resume:
${resumeText}

Format response as JSON with these keys: score, strengths, improvements, suggestions
  `.trim();

  const response = await ai.models.generateContent({
    model: "gemini-2.0-flash-001",
    contents: prompt,
  });
  if (!response) {
    throw new Error("No response text from Gemini");
  }
  const result: any = response.text;
  console.log(response.text);

  return {
    score: result?.score || 0,
    strengths: result?.strengths || [],
    improvements: result?.improvements || [],
    suggestions: result?.suggestions || [],
  };
}

File: server-2/app/utils/gemini.py
import os
import json
import re
import google.generativeai as genai
from dotenv import load_dotenv

load_dotenv()

genai.configure(api_key=os.getenv("GEMINI_API_KEY"))

def analyze_resume(resume_text):
    prompt = f"""
    Analyze this resume for ATS (Applicant Tracking System) optimization. Provide:
    1. A score out of 100
    2. Key strengths
    3. Areas for improvement
    4. Suggested improvements.
    
    All of these should be an array.
    
    Resume:
    {resume_text}
    
    Format response as JSON with these keys: score, strengths, improvements, suggestions
    """
    
    model = genai.GenerativeModel('gemini-1.5-pro')
    response = model.generate_content(prompt)
    
    try:
        # Extract the JSON content from the response text
        text_content = response.text
        
        # If the response is wrapped in markdown code block, extract just the JSON
        json_match = re.search(r'```json\s*(.*?)\s*```', text_content, re.DOTALL)
        if json_match:
            json_str = json_match.group(1)
        else:
            json_str = text_content
            
        # Parse the JSON string
        result = json.loads(json_str)
        print("Parsed JSON:", result)
        return result
    except Exception as e:
        print(f"Error parsing Gemini response: {str(e)}")
        return {"error": str(e), "score": 0, "strengths": [], "improvements": [], "suggestions": []}